### PR TITLE
Add conformance test for row_major matrices

### PR DESCRIPTION
**Pull request by JasperNV**
Requires adding some simple support for uniform blocks.

Currently breaks on some Apple platforms.

Fixes issue #2882.

### CODE REVIEW

File: sdk/tests/conformance2/glsl3/matrix-row-major.html [hunk 1]
```diff
@@ -0,0 +1,72 @@
+<!--
+
+/*
+** Copyright (c) 2015 The Khronos Group Inc.
```
Comment (kainino0x) [inline sdk/tests/conformance2/glsl3/matrix-row-major.html]: nit: 2019

File: sdk/tests/conformance2/glsl3/00_test_list.txt [hunk 2]
```diff
@@ -22,6 +22,7 @@ frag-depth.html
 invalid-default-precision.html
 invalid-invariant.html
 loops-with-side-effects.html
+matrix-row-major.html
```
Comment (kainino0x) [inline sdk/tests/conformance2/glsl3/00_test_list.txt]: nit: `--min-version 2.0.1`

**Comment by JasperNV** [comment]
Thanks for the speedy review. Fixed the nits.

**Comment by kenrussell** [comment]
Good news: this test passes on a 2017 MacBook Pro with dual Intel Graphics HD 630 and AMD Radeon Pro 560 GPUs, running 10.14.4. It seems the underlying driver bug has already been fixed.

Note that if you have additional test cases that expose bugs, please do put up new pull requests. Thanks.

**Comment by WJsjtu** [comment]
I've met a problem: If I change the matrix in [matrix-row-major](https://www.khronos.org/registry/webgl/sdk/tests/conformance2/glsl3/matrix-row-major.html?webglVersion=2&dumpShaders=undefined&quiet=0) from mat4x3 to mat4 the test fails. Is there any limits for the row_major qualifier?

**Pull request merged**